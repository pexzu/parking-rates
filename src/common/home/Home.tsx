import * as React from 'react';
import * as E from './HomeStyles';
import DateTimePicker from '../../components/dateTimePicker/DateTimePicker';
import { PrimaryButton } from '../../styled/buttons/PrimaryButton';

interface HomeProps {}
interface HomeState {
  entryDate: Date;
  exitDate: Date;
}

export default class Home extends React.Component<HomeProps, HomeState> {
  constructor(props: HomeProps) {
    super(props);
    this.state = {
      entryDate: new Date(),
      exitDate: new Date(),
    };
  }

  updateEntryDate = (date: Date) => {
    this.setState({
      entryDate: date,
    });
  };

  updateExitDate = (date: Date) => {
    this.setState({
      exitDate: date,
    });
  };

  render() {
    const { entryDate, exitDate } = this.state;
    return (
      <E.HomeContainer>
        <E.HomeHeaderBlock>Parking Rate Calculator</E.HomeHeaderBlock>
        <E.HomeInputBlock>
          <DateTimePicker
            inputLabel='Entry Date & Time'
            startDate={entryDate}
            onChange={this.updateEntryDate}
          />
          <DateTimePicker
            inputLabel='Exit Date and Time'
            startDate={exitDate}
            onChange={this.updateExitDate}
          />
          <PrimaryButton>Submit</PrimaryButton>
        </E.HomeInputBlock>
        <E.HomeInfoBlock />
      </E.HomeContainer>
    );
  }
}
