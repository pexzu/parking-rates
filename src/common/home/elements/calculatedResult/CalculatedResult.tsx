import * as React from 'react';

export interface ICalculatedResult {
  entry: Date;
  exit: Date;
}

export function CalculatedResult(props: ICalculatedResult) {
  return (
    <div>
      stst
      <div />
    </div>
  );
}
