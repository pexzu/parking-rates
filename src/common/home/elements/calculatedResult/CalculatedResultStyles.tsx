import styled from '@emotion/styled';

export const CalculatedResultContatiner = styled.div`
  label: CalculatedResultContatiner;
  padding: 15px;
  font-weight: bold;
`;
