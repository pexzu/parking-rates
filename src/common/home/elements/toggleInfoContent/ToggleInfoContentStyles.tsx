import styled from '@emotion/styled';
import { drop } from '../../HomeStyles';

export const ToggleInfoContentBlock = styled.div`
  label: ToggleInfoContentBlock;
  background: #ffffff;
  border-bottom: 1px inset;
`;

export const ToggleInfoContentBlockLabel = styled.div`
  label: ToggleInfoContentBlockLabel;
`;

export const ToggleInfoContentBlockSection = styled.div`
  label: ToggleInfoContentBlockSection;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 18px 48px;
`;

export const ToggleInfoContentBlockLeftSide = styled.div`
  label: ToggleInfoContentBlockLeftSide;
`;

export const ToggleInfoContentBlockRightSide = styled.div`
  label: ToggleInfoContentBlockRightSide;
`;

export const ToggleInfoContentBlockConditon = styled.div`
  label: ToggleInfoContentBlockConditon;
  font-size: 12px;
  text-align: center;
  padding-bottom: 3px;
`;

export const ToggleInfoContentBlockType = styled.div`
  label: ToggleInfoContentBlockType;
  font-size: 12px;
  text-align: center;
`;

export const ToggleInfoContentContainer = styled.div`
  label: ToggleInfoContentContainer;
  animation: ${drop} 0.4s linear forwards;
`;

export const ToggleInfoContentBlockPrice = styled.div`
  label: ToggleInfoContentBlockPrice;
  font-weight: bold;
  padding: 3px;
`;

export const OtherRatesContainer = styled.div`
  label: OtherRatesContainer;
`;

export const OtherRatesBlock = styled.div`
  label: OtherRatesBlock;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px inset;
`;

export const OtherRatesPeriod = styled.div`
  label: OtherRatesPeriod;
  padding: 0 6px;
`;

export const OtherRatesCost = styled(ToggleInfoContentBlockPrice)`
  label: OtherRatesCost;
  padding: 6px;
`;
