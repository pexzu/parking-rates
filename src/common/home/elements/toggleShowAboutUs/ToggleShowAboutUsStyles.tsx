import styled from '@emotion/styled';
import { drop } from '../../HomeStyles';

export const AboutUsContainer = styled.div`
  label: AboutUsContainer;
  padding: 24px 48px;
  background: #ffffff;
  border-bottom: 1px inset;
  box-shadow: 0 0 1px 0 #000000;
  animation: ${drop} 1s ease;
`;
