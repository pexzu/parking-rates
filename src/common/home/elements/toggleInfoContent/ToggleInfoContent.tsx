import * as React from 'react';
import * as E from './ToggleInfoContentStyles';
import { rates } from '../../../../mockData/mockData';

export interface IToggleInfoContentProps {}

export const ToggleInfoContent = (props: IToggleInfoContentProps) => {
  return (
    <E.ToggleInfoContentContainer>
      {rates.map((item, keyIndex) => (
        <E.ToggleInfoContentBlock key={item.name + keyIndex}>
          <E.ToggleInfoContentBlockSection>
            <E.ToggleInfoContentBlockLeftSide>
              <E.ToggleInfoContentBlockLabel>{item.name}</E.ToggleInfoContentBlockLabel>
            </E.ToggleInfoContentBlockLeftSide>
            {item.entry && item.exit && (
              <E.ToggleInfoContentBlockConditon>
                {item.entry} and {item.exit}
              </E.ToggleInfoContentBlockConditon>
            )}
            <E.ToggleInfoContentBlockRightSide>
              {typeof item.price === 'string' ? (
                <E.ToggleInfoContentBlockPrice>{item.price}</E.ToggleInfoContentBlockPrice>
              ) : (
                renderOtherRates(item.price)
              )}
              <E.ToggleInfoContentBlockType>({item.type})</E.ToggleInfoContentBlockType>
            </E.ToggleInfoContentBlockRightSide>
          </E.ToggleInfoContentBlockSection>
        </E.ToggleInfoContentBlock>
      ))}
    </E.ToggleInfoContentContainer>
  );
};

const renderOtherRates = (price: [{}]) => {
  return (
    <E.OtherRatesContainer>
      {price.map((item: any) =>
        Object.keys(item).map((key, keyIndex) => (
          <E.OtherRatesBlock key={keyIndex}>
            <E.OtherRatesPeriod> {key}</E.OtherRatesPeriod>
            <E.OtherRatesCost> {item[key]}</E.OtherRatesCost>
          </E.OtherRatesBlock>
        ))
      )}
    </E.OtherRatesContainer>
  );
};
