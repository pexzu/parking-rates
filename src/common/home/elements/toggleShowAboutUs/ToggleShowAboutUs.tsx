import * as React from 'react';
import { aboutUs } from '../../../../data/data';
import * as E from './ToggleShowAboutUsStyles';

export interface IToggleShowAboutUsProps {}

export const ToggleShowAboutUs = (props: IToggleShowAboutUsProps) => {
  return <E.AboutUsContainer>{aboutUs}</E.AboutUsContainer>;
};
