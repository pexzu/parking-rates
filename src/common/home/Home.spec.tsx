import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Home from './Home';
import { ToggleInfoContent } from './elements/toggleInfoContent/ToggleInfoContent';
import { ToggleShowAboutUs } from './elements/toggleShowAboutUs/ToggleShowAboutUs';
import { CalculatedResult } from './elements/calculatedResult/CalculatedResult';
import {
  OtherRatesBlock,
  OtherRatesContainer,
  OtherRatesCost,
  OtherRatesPeriod,
  ToggleInfoContentBlock,
  ToggleInfoContentBlockRightSide,
  ToggleInfoContentBlockConditon,
  ToggleInfoContentBlockLabel,
  ToggleInfoContentBlockLeftSide,
  ToggleInfoContentBlockPrice,
  ToggleInfoContentBlockSection,
  ToggleInfoContentBlockType,
  ToggleInfoContentContainer,
} from './elements/toggleInfoContent/ToggleInfoContentStyles';

import { AboutUsContainer } from './elements/toggleShowAboutUs/ToggleShowAboutUsStyles';
import { CalculatedResultContatiner } from './elements/calculatedResult/CalculatedResultStyles';
import {
  HomeContainer,
  HomeHeaderBlock,
  HomeInputBlock,
  HomeInputBlockMessage,
} from './HomeStyles';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('Home component', () => {
  test('render', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.exists()).toBe(true);
  });
});

describe('ToggleInfoContent component', () => {
  test('render', () => {
    const wrapper = shallow(<ToggleInfoContent />);
    expect(wrapper.exists()).toBe(true);
  });
});

describe('ToggleShowAboutUs component', () => {
  test('render', () => {
    const wrapper = shallow(<ToggleShowAboutUs />);
    expect(wrapper.exists()).toBe(true);
  });
});

describe('CalculatedResult component', () => {
  test('render', () => {
    const wrapper = shallow(<CalculatedResult entry={new Date()} exit={new Date()} />);
    expect(wrapper.exists()).toBe(true);
  });
});

//Snapshot Testing
it('HomeContainer renders correctly', () => {
  const tree = renderer.create(<HomeContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('HomeHeaderBlock renders correctly', () => {
  const tree = renderer.create(<HomeHeaderBlock />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('HomeInputBlock renders correctly', () => {
  const tree = renderer.create(<HomeInputBlock />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('HomeInputBlockMessage renders correctly', () => {
  const tree = renderer.create(<HomeInputBlockMessage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('OtherRatesBlock renders correctly', () => {
  const tree = renderer.create(<OtherRatesBlock />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('OtherRatesContainer renders correctly', () => {
  const tree = renderer.create(<OtherRatesContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('OtherRatesCost renders correctly', () => {
  const tree = renderer.create(<OtherRatesCost />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('OtherRatesPeriod renders correctly', () => {
  const tree = renderer.create(<OtherRatesPeriod />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlock renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlock />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlockRightSide renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlockRightSide />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlockConditon renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlockConditon />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlockLabel renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlockLabel />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlockLeftSide renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlockLeftSide />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentContainer renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlockType renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlockType />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlockSection renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlockSection />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ToggleInfoContentBlockPrice renders correctly', () => {
  const tree = renderer.create(<ToggleInfoContentBlockPrice />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('AboutUsContainer renders correctly', () => {
  const tree = renderer.create(<AboutUsContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('CalculatedResultContatiner renders correctly', () => {
  const tree = renderer.create(<CalculatedResultContatiner />).toJSON();
  expect(tree).toMatchSnapshot();
});
