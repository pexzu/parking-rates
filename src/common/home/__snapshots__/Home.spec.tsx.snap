// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsContainer renders correctly 1`] = `
<div
  className="css-1myhlma-AboutUsContainer"
/>
`;

exports[`CalculatedResultContatiner renders correctly 1`] = `
<div
  className="css-1nl5bh1-CalculatedResultContatiner"
/>
`;

exports[`HomeContainer renders correctly 1`] = `
<div
  className="css-bplkqs-HomeContainer"
/>
`;

exports[`HomeHeaderBlock renders correctly 1`] = `
<header
  className="css-i6ue13-HomeHeaderBlock"
/>
`;

exports[`HomeInputBlock renders correctly 1`] = `
<div
  className="css-1hicozr-HomeInputBlock"
/>
`;

exports[`HomeInputBlockMessage renders correctly 1`] = `
<div
  className="css-1pvpznp-HomeInputBlockMessage"
/>
`;

exports[`OtherRatesBlock renders correctly 1`] = `
<div
  className="css-1opqkqn-OtherRatesBlock"
/>
`;

exports[`OtherRatesContainer renders correctly 1`] = `
<div
  className="css-kx8e9j-OtherRatesContainer"
/>
`;

exports[`OtherRatesCost renders correctly 1`] = `
<div
  className="css-4v9t9r-ToggleInfoContentBlockPrice-OtherRatesCost"
/>
`;

exports[`OtherRatesPeriod renders correctly 1`] = `
<div
  className="css-ii28od-OtherRatesPeriod"
/>
`;

exports[`ToggleInfoContentBlock renders correctly 1`] = `
<div
  className="css-4a5q76-ToggleInfoContentBlock"
/>
`;

exports[`ToggleInfoContentBlockConditon renders correctly 1`] = `
<div
  className="css-19b61i6-ToggleInfoContentBlockConditon"
/>
`;

exports[`ToggleInfoContentBlockLabel renders correctly 1`] = `
<div
  className="css-1q45o0s-ToggleInfoContentBlockLabel"
/>
`;

exports[`ToggleInfoContentBlockLeftSide renders correctly 1`] = `
<div
  className="css-1wrzv2e-ToggleInfoContentBlockLeftSide"
/>
`;

exports[`ToggleInfoContentBlockPrice renders correctly 1`] = `
<div
  className="css-p82oyo-ToggleInfoContentBlockPrice"
/>
`;

exports[`ToggleInfoContentBlockRightSide renders correctly 1`] = `
<div
  className="css-ydhful-ToggleInfoContentBlockRightSide"
/>
`;

exports[`ToggleInfoContentBlockSection renders correctly 1`] = `
<div
  className="css-rflijk-ToggleInfoContentBlockSection"
/>
`;

exports[`ToggleInfoContentBlockType renders correctly 1`] = `
<div
  className="css-1ib1lc9-ToggleInfoContentBlockType"
/>
`;

exports[`ToggleInfoContentContainer renders correctly 1`] = `
<div
  className="css-1p1kq2u-ToggleInfoContentContainer"
/>
`;
